silverorange/composer-repository
# Add Forwarding to Jenkins Server

Testing Instructions:
First clone both this and the branch in this PR https://github.com/silverorange/github-webhook-handler/pull/5

Install the dependencies for both, note that in this repo's composer.json you'll need to set the github-webhook-handler version back to ^0.3.0

`rm -rf vendor/silverorange/github-webhook-handler` and then set up a symlink to the version with the newest changes `ln -s /so/packages/github-webhook-handler/work-$USER/ vendor/silverorange/github-webhook-handler`

Once that's all set up you need to make a change that can be registered by Jenkins. I've created a PR specifically for testing: https://github.com/silverorange/site/pull/226 I'd recommend just pushing to that branch with a change that would show a broken build, like adding an extra line to the end of a file.

Upon committing you now have to find the payload that would have been delivered, and send it to the local version of the webhook. To find the payload click on the Settings tab in the Site repo, then go to Webhooks, click on webhook.silverorange.com, find the most recent delivery and click on it to get the payload and headers.

Now to recreate that payload and send it to the local webhook, I recommend using [Postman](https://www.getpostman.com/). In the app create a new POST request. First copy the bottom 3 headers (User-Agent, X-GitHub-Delivery, and X-GitHub-Event) into the Headers tab. Then for the body, select form-data, set the key to "payload" and then the value to the payload JSON displayed in the recent deliveries section. The request URL bar should be set to your local version of the webhook, e.g, http://hamlin/composer-repository/work-ross/www/webhook/index.php. After all the configuration you can hit send. If there's no body in the response that's normal, but you should see Jenkins start to process your commit on the Site PR within a minute.
